source/tool/chipsec/hal/acpi_tables.py

```diff
@@ -584,3 +586,30 @@ def __str__( self ):
 ACPI Table Entries:
 %s
 """ %( ''.join( ['0x%016X\n' % addr for addr in self.Entries]))
+
+########################################################################################################
+#
+# FADT Table
+#
+########################################################################################################
+
+class FADT (ACPI_TABLE):
+    def __init__( self ):
+        self.dsdt = None
+        self.x_dsdt = None
+
+    def parse( self, table_content ):
+        self.dsdt = struct.unpack('<I', table_content[4:8])[0]
+        try:

```

Comment:
Any reason why we are not using a condition on the length of table_content to try to fill XSDT address?